File: frontEnd/lib/core/models/club.dart
// ============================================================================
// CLUB MODEL - Represents student clubs and organizations
// ============================================================================

class Club {
  final int id;
  final String name;
  final String description;
  final String? imageUrl;
  final String category;
  final int memberCount;
  final bool isMember;
  final String? president;
  final String? contactEmail;

  Club({
    required this.id,
    required this.name,
    required this.description,
    this.imageUrl,
    required this.category,
    required this.memberCount,
    required this.isMember,
    this.president,
    this.contactEmail,
  });

  factory Club.fromJson(Map<String, dynamic> json) {
    return Club(
      id: json['id'] as int,
      name: json['name'] as String,
      description: json['description'] as String,
      imageUrl: json['image_url'] as String?,
      category: json['category'] as String,
      memberCount: json['member_count'] as int,
      isMember: json['is_member'] as bool? ?? false,
      president: json['president'] as String?,
      contactEmail: json['contact_email'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'image_url': imageUrl,
      'category': category,
      'member_count': memberCount,
      'is_member': isMember,
      'president': president,
      'contact_email': contactEmail,
    };
  }
}

File: Backend/app/Models/Club.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsToMany;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\SoftDeletes;

class Club extends Model
{
    use HasFactory, SoftDeletes;

    protected $fillable = [
        'name',
        'description',
        'category',
        'member_count',
        'president_name',
        'meeting_schedule',
        'image_url',
        'is_active',
    ];

    protected $casts = [
        'member_count' => 'integer',
        'is_active' => 'boolean',
    ];

    /**
     * Get club memberships.
     */
    public function memberships(): HasMany
    {
        return $this->hasMany(ClubMembership::class);
    }

    /**
     * Get club members.
     */
    public function members(): BelongsToMany
    {
        return $this->belongsToMany(Student::class, 'club_memberships')
            ->withPivot(['join_date', 'role', 'status'])
            ->withTimestamps();
    }

    /**
     * Get active members.
     */
    public function activeMembers(): BelongsToMany
    {
        return $this->members()->wherePivot('status', 'active');
    }

    /**
     * Increment member count.
     */
    public function incrementMemberCount(): bool
    {
        $this->increment('member_count');
        return true;
    }

    /**
     * Decrement member count.
     */
    public function decrementMemberCount(): bool
    {
        if ($this->member_count > 0) {
            $this->decrement('member_count');
        }
        return true;
    }
}
